### DISCUSSION TITLE
Can we use one publisher and few subscribers but both subscribers having different data members.

### DISCUSSION THREAD

**Discussion by Akul1903** [eProsima/Fast-RTPS-docs#901] [General]
Kindly help its urgent

while working on fast dds publisher subscriber model , I get stuck in a problem
I am working on one publisher and 2 subscriber model in which each subscriber has different data members.

For example:

Publisher idl file:
enum Status {
    ACTIVE,
    INACTIVE,
    PENDING
};

struct user {
    char id;
    string name;
    string email;
};

struct helloworld {
    unsigned long index;
    string message;
    Status ss; 
    user userInfo; 
};

Subscriber 1 idl file:
struct user {
    char id;
    string name;
    string email;
};

struct helloworld {
    unsigned long index;
    string message;
    user userInfo; 
};

and subscriber 2 idl file:
enum Status {
    ACTIVE,
    INACTIVE,
    PENDING
};

struct helloworld {
    unsigned long index;
    string message;
    Status ss; 
};

SO BASICALLY THIS IS MY QUESTION , CAN IT BE POSSIBLE IF YES,
KINDLY PROVIDE SOME SOLUTION TO GO AHEAD.

**Comment by JesusPoderoso** [comment]
Closing this issue in favor of
- https://github.com/eProsima/Fast-DDS-docs/discussions/900
